application/console/migrations/m170203_210542_create_initial_tables.php

```diff
@@ -39,38 +42,44 @@ private function createUserTable()
             "ENGINE=InnoDB DEFAULT CHARSET=utf8"
         );
 
-        $this->createIndex('uq_user_employee_id','{{user}}','employee_id',true);
-        $this->createIndex('uq_user_username','{{user}}','username',true);
-        $this->createIndex('uq_user_email','{{user}}','email',true);
+        $this->createIndex('uq_user_employee_id', '{{user}}', 'employee_id', true);
+        $this->createIndex('uq_user_username', '{{user}}', 'username', true);
+        $this->createIndex('uq_user_email', '{{user}}', 'email', true);
     }
 
     private function dropUserTable()
     {
         $this->dropTable('{{user}}');
     }
 
-    private function createPasswordHistoryTable()
+    private function createPasswordTable()
     {
         $this->createTable(
-            '{{password_history}}',
+            '{{password}}',
             [
                 'id' => 'pk',
                 'user_id' => 'int(11) not null',
-                'password_hash' => 'varchar(255) not null',
+                'hash' => 'varchar(255) not null',
                 'created_utc' => 'datetime not null',
+                'expiration_utc' => 'datetime not null',
+                'grace_period_ends_utc' => 'datetime not null',
             ],
             "ENGINE=InnoDB DEFAULT CHARSET=utf8"
         );
+    }
 
-        $this->createIndex('idx_password_hash','{{password_history}}','password_hash',false);
-
-        $this->addForeignKey('fk_password_history_user_id','{{password_history}}','user_id','{{user}}','id','CASCADE','CASCADE');
+    private function dropPasswordTable()
+    {
+        $this->dropTable('{{password}}');
     }
 
-    private function dropPasswordHistoryTable()
+    private function createForeignKeys()
     {
-        $this->dropForeignKey('fk_password_history_user_id','{{password_history}}');
+        $this->addForeignKey('fk_user_to_current_password','{{user}}','password_id','{{password}}','id','CASCADE','CASCADE');
```

Comment:
I don't think there's a use case for it @fillup but can certainly add it easy enough.  The only thing it might end up doing for us in the long run is frustrating us if we're not removing rows in the right order...

Agreed @forevermatt , if we do want it, we don't want users being removed when a password is removed :-)